### DISCUSSION TITLE
WIFI Access Point on Yi Home Camera

### DISCUSSION THREAD

**Discussion by spsi22** [roleoroleo/yi-hack-Allwinner-v2#361] [Ideas]
Is it possible to setup the Yi Home camera with its own Access Point mode using this hack?
ie where you can connect directly to the camera from your phone via WIFI.

**Comment by roleoroleo** [comment]
It's not possible at the moment.

**Reply by spsi22** [reply]
Ok, is it possible for the camera to achieve this with its hardware?
If, so are there any plans to make this a feature on the hack?
I could really do with this for remote access to the camera - no internet/router etc needed

**Reply by roleoroleo** [reply]
> Ok, is it possible for the camera to achieve this with its hardware?

Yes, I think it is enough to change the configuration files.

> If, so are there any plans to make this a feature on the hack?
> I could really do with this for remote access to the camera - no internet/router etc needed

At the moment is not planned.

**Comment by spsi22** [comment]
Is it possible you and the team could release some simple steps on how I could possibly implement this?

We're wanting to make the camera usable in areas where there is no internet for monitoring our baby - use it like a baby monitor.

**Reply by roleoroleo** [reply]
I checked the original fw and I think dhcp server is missing.
Probably you could configure the cam as a wifi access point but I think it's not able to generate an IP address for the connecting client.

If you want to try the 1st step, you should try to configure wpa_supplicant.conf for AP mode and put the file on the SD card following this post:
https://github.com/roleoroleo/yi-hack-Allwinner/issues/289

About 2nd step, I will try to compile udhcpd.

**Reply by spsi22** [reply]
Thanks! I'm not a coder or anything so I'm learning as I go here!
Regarding the file I'm not totally sure I understand what the function is of wpa_supplicant.conf.
Are you meant to add a wifi network (SSID & pwd) to this file? If so, how does this file help achieve AP mode (on the camera) when you are in an area where you do not have an internet connection / wifi?

Also just another thought, is it possible to use an ESP8266 as the wifi access point to connect to the camera and then access this with the phone/computer instead? If so, what steps would you need to take here? Or is this too complicated? 
My thinking for this is that it could actually work really well at extending the range between two connections also.

**Reply by roleoroleo** [reply]
> Thanks! I'm not a coder or anything so I'm learning as I go here! Regarding the file I'm not totally sure I understand what the function is of wpa_supplicant.conf. Are you meant to add a wifi network (SSID & pwd) to this file? If so, how does this file help achieve AP mode (on the camera) when you are in an area where you do not have an internet connection / wifi?

No luck with my tests.
I'm not able to start a wifi access point with a custom configuration file.
Because yi software restarts the daemon every time I kill it.
And I can't kill yi software.

> Also just another thought, is it possible to use an ESP8266 as the wifi access point to connect to the camera and then access this with the phone/computer instead? If so, what steps would you need to take here? Or is this too complicated? My thinking for this is that it could actually work really well at extending the range between two connections also.

I think it's possible but I'm not a ESP programmer. Maybe other users could help you.
